tasmota/tasmota_support/support_tasmota.ino

```diff
@@ -392,13 +392,17 @@ void SetAllPower(uint32_t state, uint32_t source)
     power_t all_on = POWER_MASK >> (POWER_SIZE - TasmotaGlobal.devices_present);
     switch (state) {
     case POWER_OFF:
-      TasmotaGlobal.power = 0;
-      break;
+        // keep loocked bits and set all other to 0
+        TasmotaGlobal.power &= Settings->power_lock; 
```

Comment:
This will turn on relays which have locked bit set but were OFF. Not good.

Ditto for the other changes.

Reply:
ok readded the all_on filter and do some testing. Looks reasonable now to me

Reply:
Regarding: TasmotaGlobal.power &= Settings->power_lock;  Your finding is wrong. it is a &= and therefore keep only the locked relays ON. All other go OFF as intended.